library/common/engine.h

```diff
@@ -39,8 +39,9 @@ class Engine {
    * Increment a counter with a given string of elements and by the given count.
    * @param elements, joined elements of the timeseries.
    * @param count, amount to add to the counter.
+   * @param envoy_status_t, the resulting status of the operation.
    */
-  void recordCounter(const std::string& elements, uint64_t count);
+  envoy_status_t recordCounter(const std::string& elements, uint64_t count);
```

Comment:
Since we're here let's add the `envoy_engine_t` argument as well.

Reply:
I wanted to be more selective in this PR and then have that in a subsequent one. wdyt @goaway?